inference-engine/tests/functional/inference_engine/transformations/convert_previous_nms_to_nms_5.cpp

```diff
@@ -0,0 +1,234 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <string>
+#include <memory>
+#include <queue>
+
+#include <ngraph/function.hpp>
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/opsets/opset3.hpp>
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/opsets/opset5.hpp>
+#include <transformations/op_conversions/convert_previous_nms_to_nms_5.hpp>
+#include <transformations/init_node_info.hpp>
+#include <transformations/utils/utils.hpp>
+#include <ngraph/pass/manager.hpp>
+
+#include "common_test_utils/ngraph_test_utils.hpp"
+
+using namespace testing;
+using namespace ngraph;
+
+TEST(TransformationTests, ConvertNMS4FiveInputsToNMS5) {
```

Comment:
Add some tests when not all optional inputs of the original NMS operation are connected.

Reply:
There are such tests: `ConvertNMS1TwoInputsToNMS5`, `ConvertNMS3TwoInputsToNMS5`,  `ConvertNMS4TwoInputsToNMS5`. And, according to the file `ngraph/core/include/ngraph/op/non_max_suppression.hpp`, NMS-1, NMS-3, NMS-4 have only constructors with 2 or 5 inputs.